### PR TITLE
fix: binding dependencies on filter calculated variables

### PR DESCRIPTION
**Pull request by davdarras**

### CODE REVIEW

File: eno-core/src/test/java/fr/insee/eno/core/mapping/out/lunatic/ComponentFilterTest.java [hunk 1]
```diff
@@ -46,22 +45,6 @@ void defaultComponentFilter() {
...
-        //
-        LunaticMapper lunaticMapper = new LunaticMapper();
-        lunaticMapper.mapEnoObject(enoComponentFilter, lunaticConditionFilter);
-        //
-        assertEquals("(FOO_VARIABLE = 1)", lunaticConditionFilter.getValue());
-        assertEquals(Constant.LUNATIC_LABEL_VTL, lunaticConditionFilter.getType());
-        assertEquals(1, lunaticConditionFilter.getBindingDependencies().size());
-        assertEquals("FOO_VARIABLE", lunaticConditionFilter.getBindingDependencies().get(0));
-    }
-
```
Comment (nsenave) [inline eno-core/src/test/java/fr/insee/eno/core/mapping/out/lunatic/ComponentFilterTest.java:64]: 😢

File: eno-core/src/main/java/fr/insee/eno/core/model/navigation/ComponentFilter.java [hunk 2]
```diff
@@ -39,14 +39,17 @@ public class ComponentFilter extends EnoObject {
     @Getter
     private final Set<BindingReference> bindingReferences = new HashSet<>();
 
+    @Getter
+    private final Set<BindingReference> resolvedBindingReferences = new HashSet<>();
```
Comment (nsenave) [inline eno-core/src/main/java/fr/insee/eno/core/model/navigation/ComponentFilter.java:43]: This attribute is introduced to make a proper separation between "unresolved" vs "resolved" dependencies? (i.e. to have less mutations) If so, I think the same idea should be applied in the `CalculatedExpression` object which has the same pattern (initial bindigReferences set which is updated ("resolved") in a processing)

Comment (nsenave) [reply eno-core/src/main/java/fr/insee/eno/core/model/navigation/ComponentFilter.java:43]: \+ maybe a line of javadoc on this attribute (or both attributes) would be helpful

Comment (davdarras) [reply eno-core/src/main/java/fr/insee/eno/core/model/navigation/ComponentFilter.java:43]: You're right, we need unresolved dependencies in filter result calculated variables, so i introduced the resolved depencies attributes for condition filter. I'm not sure about introducting the same pattern to CalculatedExpression, as unresolved dependencies are not needed. Maybe we should rename the attribute as resolvedDependencies instead ?

**Pull request merged**